lib/identity_cache/query_api.rb

```diff
@@ -246,96 +272,102 @@ def find_batch(ids)
         ids.map{ |id| records_by_id[id] }
       end
 
-      def prefetch_associations(associations, records)
-        associations = hashify_includes_structure(associations)
+      def fetch_embedded_associations(records)
+        associations = embedded_associations
+        return true if associations.empty?
 
-        associations.each do |association, sub_associations|
-          case
-          when details = cached_has_manys[association]
+        return false unless primary_cache_index_enabled
 
-            if details[:embed] == true
-              child_records = records.map(&details[:cached_accessor_name].to_sym).flatten
+        records_by_id = records.index_by(&:id)
+        cached_records = fetch_multi(records.map(&:id))
```

Comment:
not sure this line makes sense in the context of the change - isn't the whole point that if we were passed in a complete record already (e.g. one fetched straight from the DB in an update scenario) that we *don't* want to go to the cache for that record because it could have incorrect associations, and we just want to go the cache for the associations of the "concrete" record we already have?